app/addons/fauxton/navigation/stores.js

```diff
@@ -94,8 +107,7 @@ Stores.NavBarStore = FauxtonAPI.Store.extend({
   },
 
   toggleMenu () {
-    app.utils.localStorageSet(FauxtonAPI.constants.LOCAL_STORAGE.SIDEBAR_MINIMIZED,
-                              !this.isMinimized());
+    this._isMinimized = !this._isMinimized;
```

Comment:
I think I'd prefer to leave this as-is.  The new logic forces the sidebar to be minimized by default, showing users that they have much more screen real estate to work with, which is inline with the user research and feedback.